lib/internal/modules/esm/get_format.js

```diff
@@ -36,26 +38,27 @@ if (experimentalWasmModules)
 if (experimentalJsonModules)
   extensionFormatMap['.json'] = legacyExtensionFormatMap['.json'] = 'json';
 
-function defaultGetFormat(url, context, defaultGetFormatUnused) {
-  if (StringPrototypeStartsWith(url, 'node:')) {
-    return { format: 'builtin' };
-  }
-  const parsed = new URL(url);
-  if (parsed.protocol === 'data:') {
+const protocolHandlers = ObjectCreate(null);
+
+ObjectAssign(protocolHandlers, {
```

Comment:
It's written this way because it [needs a null prototype](https://github.com/nodejs/node/pull/37468#discussion_r691379026), they need to be own methods rather than properties with functional values.

I broke it up to make it seemed bitter isolation and cleaner.

Would you prefer this:

```js
const protocolHandlers = ObjectAssign(ObjectCreate(null), {
```

The result is the same.

Reply:
It doesn’t need to change, it just took me several passes before I realized what syntax I was reading. I don’t encounter objects of methods often. I was more thinking that for future readability it would be better to use a more common pattern; but I guess the only other option is a long `if` and I can see why you’d prefer something encapsulated over that.

@bmeck can you explain when/why something needs a null prototype? We declare literal objects all the time in the code, I thought: `const protocolHandlers = { method(): {} }`.

Reply:
whenever you have something used as a dictionary with dynamic property access, in this example if some horrible code returned `toString` during the lookup for example it would run `Object.prototype.toString` as the handler. In theory this should error, but it might not and means that people could inject handlers onto `Object.prototype` so that things like `Object.prototype['ftp:']` is seen as a protocol handler.

Reply:
If only the bug preventing a class extending null would ever get fixed 😢

I think the `const protocolHandlers = ObjectAssign(ObjectCreate(null), {` is a bit cleaner, and aside from the first line, it'll look exactly the same.